### DISCUSSION TITLE
Renovate uses invalid php version although it is set in composer.json

### DISCUSSION THREAD

**Discussion by joey-bolts** [renovatebot/renovate#19376] [Request Help] [answered]
Just tried to run renovate. My current strategy just updates the composer.lock with the latest patch version. 
- v10.4.30 -> v10.4.32 

This works like a charm, and it updates with dependencies. This also works like a charm except it does not see i should use php7.4 for this instead of 8.1.

My composer.json of the project contains:
```
	"config": {
		"bin-dir": ".",
		"platform": {
			"php": "7.4"
		},
	},
```

config.js

```
// Example config.js from https://github.com/renovatebot/renovate/blob/main/docs/usage/examples/self-hosting.md
module.exports = {
    endpoint: 'https://gitlab.xxx.it/api/v4/',
    token: '**gitlab_token**',
    platform: 'gitlab',
    baseBranches: ["master"],
    enabledManagers: "composer",
    onboardingConfig: {
        extends: [
            "github>user/renovate-config",
        ],
    },
    repositories: [
        {
            repository: 'CMS/projects/myrepo/my-first-repo'
        },
        {
            repository: 'CMS/projects/myrepo/my-second-repo'
        }
    ],
};

and my config set.

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":disableDependencyDashboard",
    ":separatePatchReleases"
  ],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "baseBranches": [
    "master"
  ],
  "enabledManagers": [
    "composer"
  ],
  "separateMinorPatch": true,
  "commitMessagePrefix": "[TASK] ",
  "commitMessageTopic": "{{depName}}",
  "commitMessageExtra": " ({{{displayFrom}}} => {{{displayTo}}})",
  "rangeStrategy": "update-lockfile",
  "packageRules": [
    {
      "matchUpdateTypes": [
        "minor",
        "pin",
        "digest",
        "major"
      ],
      "enabled": false
    },
    {
      "matchUpdateTypes": [
        "patch"
      ],
      "matchPackagePatterns": [
        "*"
      ],
      "excludePackagePatterns": ["^typo3/cms-"],
      "enabled": false
    },
    {
      "matchUpdateTypes": [
        "patch"
      ],
      "matchPackagePrefixes": [
        "typo3/cms-"
      ],
      "excludePackageNames": [
        "typo3/cms-cli",
        "typo3/cms-composer-installers"
      ],
      "groupName": "TYPO3 CMS",
      "labels": [
        "typo3"
      ],
      "enabled": true
    }
  ]
}
```
The issue composer install gave on the created MR.

```
  Problem 1
    - psr/cache is locked to version 3.0.0 and an update of this package was not requested.
    - psr/cache 3.0.0 requires php >=8.0.0 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 2
    - symfony/cache is locked to version v6.2.0 and an update of this package was not requested.
    - symfony/cache v6.2.0 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 3
    - symfony/cache-contracts is locked to version v3.2.0 and an update of this package was not requested.
    - symfony/cache-contracts v3.2.0 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 4
    - symfony/deprecation-contracts is locked to version v3.2.0 and an update of this package was not requested.
    - symfony/deprecation-contracts v3.2.0 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 5
    - symfony/event-dispatcher is locked to version v6.2.0 and an update of this package was not requested.
    - symfony/event-dispatcher v6.2.0 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 6
    - symfony/string is locked to version v6.2.0 and an update of this package was not requested.
    - symfony/string v6.2.0 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 7
    - symfony/var-exporter is locked to version v6.2.1 and an update of this package was not requested.
    - symfony/var-exporter v6.2.1 requires php >=8.1 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
  Problem 8
    - psr/cache 3.0.0 requires php >=8.0.0 -> your php version (7.4; overridden via config.platform, actual: 7.4.22) does not satisfy that requirement.
    - doctrine/annotations 1.14.1 requires psr/cache ^1 || ^2 || ^3 -> satisfiable by psr/cache[3.0.0].
    - doctrine/annotations is locked to version 1.14.1 and an update of this package was not requested.
```

Is there any other way where i need to set the php version?

**Comment by joey-bolts** [comment]
As mentioned here, https://github.com/renovatebot/renovate/issues/2355#issuecomment-1293579145 config.platform.php should be respected right? We using the latest renovate image `renovate/renovate:latest`

**Comment by joey-bolts** [comment, marked as answer]
Setting the following in gitlab-ci.yml solved this issue, this is however not recommended by renovate itself. But for our use-case it is. 

```
  variables:
    RENOVATE_COMPOSER_IGNORE_PLATFORM_REQS: 'false'
```

**Comment by deviantintegral** [comment]
We're also seeing this. Our composer.json specifies php as `~8.1.0`, but renovate is using PHP 8.2. The docs indicate at https://docs.renovatebot.com/configuration-options/#composerignoreplatformreqs that the PHP version should be autodetected from composer.json.

We're using GitHub, so I think we can work around this by adding this to renovate.json:


```
  "composerIgnorePlatformReqs": [
       "php",
       "ext-*",
       "lib-*"
   ],
```

However, I think there may still be a bug in renovate selecting the wrong PHP version.
